### PR TITLE
feat: add SwitchProps

### PR DESCRIPTION
**Pull request by abhijitxy**
- [x] Added SwitchProps to standardize props for the Switch component

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -138,6 +138,21 @@ export const ledProps = commonComponentProps.extend({
...
+  closed: z.boolean().optional(),
+})
+export const switchPins = z.object({
+  p1: z.object({
+    portNumber: z.number(),
+  }),
+  p2: z.object({
+    portNumber: z.number(),
+  }),
+})
```
Comment (seveibar) [inline lib/index.ts:152]:
```suggestion
```

File: lib/index.ts [hunk 2]
```diff
@@ -138,6 +138,21 @@ export const ledProps = commonComponentProps.extend({
 export const ledPins = lrPolarPins
 export type LedProps = z.input<typeof ledProps>
 
+export const switchProps = commonComponentProps.extend({
+  ftype: z.literal("switch"),
+  closed: z.boolean().optional(),
+})
```
Comment (seveibar) [inline lib/index.ts:144]:
```suggestion
export const switchProps = commonComponentProps.extend({
  ftype: z.literal("switch"),
  switchType: z.enum(["spst"]).default("spst"),
  isNormallyClosed: z.boolean().default(false)
})
```

File: lib/index.ts [hunk 3]
```diff
@@ -138,6 +138,21 @@ export const ledProps = commonComponentProps.extend({
...
+export const switchPins = z.object({
+  p1: z.object({
+    portNumber: z.number(),
+  }),
+  p2: z.object({
+    portNumber: z.number(),
+  }),
+})
+export type SwitchProps = z.input<typeof switchProps>
+export type SwitchPins = z.input<typeof switchPins>
```
Comment (seveibar) [inline lib/index.ts:154]:
```suggestion
```

**Review by seveibar — CHANGES_REQUESTED** [review]
There are many kinds of switches, SPST, SPDT, DPDT etc. so we want to be forward looking in our API design. I've added some suggestions that I believe should cover the future cases

**Review by seveibar — CHANGES_REQUESTED** [review]
saw the re-request for review, but i still have a pending suggestion where switchType is introduced

**Comment by abhijitxy** [comment]
> There are many kinds of switches, SPST, SPDT, DPDT etc. so we want to be forward looking in our API design. I've added some suggestions that I believe should cover the future cases

 SPST is currently important as mentioned in the issue!!

**Review by seveibar — APPROVED** [review]

**Comment by seveibar** [comment]
@abhijitxy can you try to get the formatting pass to check?

**Pull request merged**